erpnext/accounts/doctype/gl_entry/gl_entry.py

```diff
@@ -63,6 +62,9 @@ def check_mandatory(self):
 
 	def pl_must_have_cost_center(self):
 		if frappe.db.get_value("Account", self.account, "report_type") == "Profit and Loss":
+			if not self.cost_center:
+				self.cost_center = erpnext.get_default_cost_center(self.company)
```

Comment:
I don't think it is a good idea to set a default cost center if not specified in the form. We should force users to set cost center in the transactions itself. We are already setting default cost center there while create a new transaction.

In current problem's context, we should edit the error message to direct user to take necessary action. In this case, PR should be amended and cost center should be entered there.

In the below message, we can change like:

```frappe.throw(_("{0} {1}: Cost Center is required for 'Profit and Loss' account {2}. Please edit/amend the {3} to set the Cost Center.")
  					.format(self.voucher_type, self.voucher_no, self.account, self.voucher_type))
```